gobblin-metastore/src/main/java/gobblin/metastore/FsStateStore.java

```diff
@@ -79,11 +79,7 @@ public FsStateStore(FileSystem fs, String storeRootDir, Class<? extends State> s
   public boolean create(String storeName)
       throws IOException {
     Path storePath = new Path(this.storeRootDir, storeName);
-    if (this.fs.exists(storePath)) {
-      throw new IOException(String.format("Store directory %s already exists for store %s", storePath, storeName));
-    }
-
-    return this.fs.mkdirs(storePath);
+    return this.fs.exists(storePath) || this.fs.mkdirs(storePath);
```

Comment:
nit:I don't think the first condition (exists) is necessary. mkdirs() will not fail if the path exists.